documents/coredocument.go

```diff
@@ -528,3 +537,22 @@ func (cd *CoreDocument) Signatures() (signatures []coredocumentpb.Signature) {
 
 	return signatures
 }
+
+func populateVersions(cd *coredocumentpb.CoreDocument, prevCD *coredocumentpb.CoreDocument) (err error) {
+	if prevCD != nil {
+		cd.PreviousVersion = prevCD.CurrentVersion
+		cd.CurrentVersion = prevCD.NextVersion
+		cd.CurrentPreimage = prevCD.NextPreimage
+	} else {
+		cd.CurrentPreimage, cd.CurrentVersion, err = utils.GenerateHashPair(idSize)
+		cd.DocumentIdentifier = cd.CurrentVersion
+		if err != nil {
+			return err
+		}
+	}
+	cd.NextPreimage, cd.NextVersion, err = utils.GenerateHashPair(idSize)
```

Comment:
I don't understand the method name. Why it is a hash pair?
Why not two methods GeneratePreImage() and GenerateNextVersion(preImage)?

Reply:
its just easier for our purpose